Suggested changes for Flavor, Image, and Network checks

vhe182 opened the pull request:
In response to a comment made by noahl, I have moved made a change.  This is to prevent needless cmd preparation when the sec_group_rule already exists.

vhe182 committed:
Move sec_group_rule to before the cmd prep

vhe182 commented:
test this please

vhe182 committed:
openstack: Move msg string preparation into the join call

meteorfox commented:
test this please

meteorfox commented:
test this please

meteorfox commented:
test this please

meteorfox commented:
test this please

vhe182 committed:
Move command issue and error raise into a new method.

There is duplication of code that issues a command and raises an error
message.  The goal is to negate this duplication. The three methods
affected are:

* _CheckFlavor
* _CheckImage
* _CheckNetworkExists

The new method created is:

* _IssueCommandCheck

vhe182 committed:
Move command issue and error raise into a new method.

There is duplication of code that issues a command and raises an error
message.  The goal is to negate this duplication. The three methods
affected are:

* _CheckFlavor
* _CheckImage
* _CheckNetworkExists

The new method created is:

* _IssueCommandCheck

vhe182 committed:
Merge branch 'openstack-cli-provider' of github.com:vhe182/PerfKitBenchmarker into openstack-cli-provider

vhe182 committed:
openstack: Move custom message creation

The custom message string is repeated multiple times.  To minimize this
duplication of code, the instruction was moved into a method.  This
method now creates the message, issues the command, and raises the
error message, if needed.

meteorfox commented on perfkitbenchmarker/providers/openstack/os_virtual_machine.py:
```diff
@@ -221,12 +213,25 @@ def _CheckFloatingIPNetworkExists(self):
-                      'run "openstack network list".',))
       raise errors.Config.InvalidValue(msg)
+    return stdout
+
```
nitpick: Remove empty line. Methods are separated by one-line

vhe182 committed:
Remove extra newline from end of method

meteorfox commented:
@vhe182 
Tested happy path and it works. :white_check_mark: 

Test for invalid machine_type=m1.smaller returns incorrect error message. :x: 
Says "Flavor" while it should've been "Machine type"

```
./pkb.py --cloud=OpenStack --machine_type=m1.smaller \
         --image='05c07bf1-59be-453a-b9d7-dd4dcc28a6b8'
         --openstack_network=private --openstack_floating_ip_pool=public \
         --benchmarks=cluster_boot --num_vms=1
```

Returns expected error message:

```
  File "/opt/PerfKitBenchmarker/perfkitbenchmarker/providers/openstack/os_virtual_machine.py", line 178, in _CheckFlavor
    self._IssueCommandCheck(cmd, 'Flavor', self.machine_type)
  File "/opt/PerfKitBenchmarker/perfkitbenchmarker/providers/openstack/os_virtual_machine.py", line 232, in _IssueCommandCheck
    raise errors.Config.InvalidValue(msg)
InvalidValue: Flavor m1.smaller could not be found. For valid flavor run "openstack flavor list".
```

Please fix with another commit, do not amend this one. Make sure other error messages are correct as well. :)

vhe182 commented:
I'm on it.  Thanks for pointing that out.

vhe182 committed:
openstack: Fix creation of wrong error message

The error message created would print a message for a flavor using the
string 'flavor' when the correction message should state 'machine
type'. To prevent this, a check is made for the type of check being
performed and the correct word is used for the flavor case.

vhe182 committed:
Fix incorrect indexing.

vhe182 committed:
openstack: Remove "," from error message

vhe182 committed:
openstack: Fix variable check error.

The condition is supposed to check the name of the check and not the
id.  Previously, the check would never trigger and the incorrect
keyword would be placed in the error message.

meteorfox commented on perfkitbenchmarker/providers/openstack/os_virtual_machine.py:
```diff
@@ -221,12 +213,28 @@ def _CheckFloatingIPNetworkExists(self):
                                        self.network_name)
+    self._IssueCommandCheck(cmd, 'Network', self.network_name)
+
+  def _IssueCommandCheck(self, cmd, name, id):
```
use a different name for the `id` parameter since it overshadows the `id()` function.

vhe182 committed:
openstack: Change name of variable

Prevent method overshadowing by changing variable name.

meteorfox commented:
Thanks!
